### ISSUE TITLE
Open source licensing

### ISSUE THREAD

**Issue by grokpot** [grokpot/Routy#94]
I don't have a clue about open source licensing, but we should probably include some sort of text telling people not to steal our shit.

Regarding Analytics: 
https://developers.google.com/analytics/devguides/collection/protocol/policy

Eric - you down to tackle this?

**Comment by ericthompson** [comment]
Definitely.

**Comment by jtran19** [comment]
Are we using anything that is licensed? If not can we Creative Commons that
ish?

On Apr 7, 2013, at 1:51 PM, ericthompson notifications@github.com wrote:

Definitely.

On Apr 7, 2013, at 1:13 PM, Ryan Prater notifications@github.com wrote:

> I don't have a clue about open source licensing, but we should probably
> include some sort of text telling people not to steal our shit.
> 
> Regarding Analytics:
> 
> https://developers.google.com/analytics/devguides/collection/protocol/policy
> 
> Eric - you down to tackle this?
> 
> —
> Reply to this email directly or view it on GitHub.

—
Reply to this email directly or view it on
GitHubhttps://github.com/ryanprater/Routy/issues/94#issuecomment-16020942
.

**Comment by grokpot** [comment]
???? idk

**Comment by ericthompson** [comment]
OK, so here's what we need to get straight.

What do we plan to do with little Routy?

Make money, cash out, and live in the hills?

Or do we want to show this bitch off and not have anyone steal it and pretend like we're saints?

Creative Commons is for [artwork](http://wiki.creativecommons.org/FAQ#Can_I_use_a_Creative_Commons_license_for_software.3F) - think books, music, that sorta thing. We can use it on the name Routy and the doodles/overall art style of the app. Not super important, and this shit is copyrighted anyways by virtue of it existing - so we're technically already covered.

We can choose to go open source with the OSI (Open Source Initiative) and one of their available [licenses](http://opensource.org/licenses) - but that's interesting in and of itself and requires us understanding that we _must_ make our code available to modification.

**Or** we can try to prevent the source code from being viewed. We can use [ProGuard](http://developer.android.com/tools/help/proguard.html) to make our shit all obfuscated and hard to read so that it will be difficult to reverse engineer it. We can also use the [Google Play Licenses](http://developer.android.com/google/play/licensing/overview.html) like [LVL](http://developer.android.com/google/play/licensing/overview.html#LVL) to have our app "phone home" every so often and verify that the app is licensed to that user. All of this stuff is built into android (sort of).

So what do we want to do?

**Comment by jtran19** [comment]
I personally like the idea of putting it up on Google Play for free for now, and then rolling out a 10-dest version for a small fee.  Or we could even restrict it to 3 destinations now and roll out a paid 5-7 destination version -- reasoning here is performance and n-fac-fucking-torial.  As a "Routy-labs" kinda thing once we've got it in the store, I'd like to play around with different techniques of honing performance.

Bottom line == if we EVER plan on monetizing, we should lock it down starting now.  I don't think it'd be a bad idea to at least make the repo private for now because we can always publicize it later if we decide post store-launch that we're just gonna offer it up for the bettering of society.

Ryan?

Jeff

**Comment by ericthompson** [comment]
Agreed with your statement. If we're going to make the repo private, can we also use the [ProGuard](http://developer.android.com/tools/help/proguard.html) thing? This way we can obfuscate the code in the event somewhat tries to reverse engineer.

The section I've read essentially says it's invoked automatically when you build the app in release mode. It's probably worth everyone reading over that page about ProGuard - it's really short.

Also, do we want to do licensing so that we can be sure each Routy instance is licensed to the individual that's using it?

**Comment by jtran19** [comment]
I thought releasing/downloading through Google Play Store automatically does a "user licensing" kind of thing.  That's how it enables the new Google Play Services API.  Let's look into this before we try to roll our own licensing system.

**Comment by grokpot** [comment]
Sorry I've been afk.

Honestly I just want to get this bitch out. You guys don't have Android
phones and we've made minimal improvements since last semester (and you
both go full time this summer).

Google takes 30%, we'd have to split the remaining 60%. We'd probably sell
it for $0.99, which means we get ~$0.20 for every download.

No one I show it to seems to _really_ dig it (just my observation) in the
sense of purchasing it (except the realty market), and if we charge people
for it, then they have a feeling of entitlement to updates/fixes/etc.

I'm more in it for the scenario of "Hey, I've got an app on the market,
it's free and you might like it".

BUT, if you guys really want to crank down on this before you graduate and
get jobs, and work out some sort of revenue model, then I'm never opposed
to beer money.

**Comment by jtran19** [comment]
I agree. I'll be full-time in Austin this summer, so I'm down to continue
supporting/improving Routy. My main goal is not to strike gold, but if
someone's offering me money I won't say no...usually.

On Apr 18, 2013, at 3:17 PM, Ryan Prater notifications@github.com wrote:

Sorry I've been afk.

Honestly I just want to get this bitch out. You guys don't have Android
phones and we've made minimal improvements since last semester (and you
both go full time this summer).

Google takes 30%, we'd have to split the remaining 60%. We'd probably sell
it for $0.99, which means we get ~$0.20 for every download.

No one I show it to seems to _really_ dig it (just my observation) in the
sense of purchasing it (except the realty market), and if we charge people
for it, then they have a feeling of entitlement to updates/fixes/etc.

I'm more in it for the scenario of "Hey, I've got an app on the market,
it's free and you might like it".

BUT, if you guys really want to crank down on this before you graduate and
get jobs, and work out some sort of revenue model, then I'm never opposed
to beer money.

On Thu, Apr 18, 2013 at 9:44 AM, jtran19 notifications@github.com wrote:

> I thought releasing/downloading through Google Play Store automatically
> does a "user licensing" kind of thing. That's how it enables the new
> Google
> Play Services API. Let's look into this before we try to roll our own
> licensing system.
> 
> On Apr 18, 2013, at 9:39 AM, ericthompson notifications@github.com
> wrote:
> 
> > Agreed with your statement. If we're going to make the repo private,
> > can
> > we also use the ProGuard thing? This way we can obfuscate the code in the
> > event somewhat tries to reverse engineer.
> > 
> > The section I've read essentially says it's invoked automatically when
> > you build the app in release mode. It's probably worth everyone reading
> > over that page about ProGuard - it's really short.
> > 
> > Also, do we want to do licensing so that we can be sure each Routy
> > instance is licensed to the individual that's using it?
> > 
> > —
> > Reply to this email directly or view it on GitHub.
> 
> —
> Reply to this email directly or view it on GitHub<
> https://github.com/ryanprater/Routy/issues/94#issuecomment-16580656>
> .

—
Reply to this email directly or view it on
GitHubhttps://github.com/ryanprater/Routy/issues/94#issuecomment-16604944
.
